src/main/java/io/pinecone/configs/PineconeConnection.java

```diff
@@ -38,27 +37,28 @@ public class PineconeConnection implements AutoCloseable {
     private VectorServiceGrpc.VectorServiceFutureStub futureStub;
 
     public PineconeConnection(String apiKey, String indexName) {
-        this.clientConfig = new PineconeClientConfig().withApiKey(apiKey);
+        this.config = new PineconeConfig(apiKey);
```

Comment:
It seems weird to me that we would build a configuration object here; as we add features to the SDK, there could be a lot of other fields added to `PineconeConfig` that we would not have correct values of when building it here. For instance, if we implemented handling of custom SSL certs, you would not ever want to be newing up a fresh config object here in the middle of of the implementation. So I think we should strive to have the user always be the one to create that config object and pass it into our implementation.

So I think this should perhaps be changed to:

```java
public PineconeConnection(PineconeConfig config, String indexName) {
  this.config = config;
  String host = getHost(config.getApiKey(), indexName)
  channel = buildChannel(host);
  initialize();
}
```

I think `PineconeConnection` should probably also have a constructor that accepts `PineconeConfig` object as the sole parameter for cases where the user knows the host or has provided a ManagedChannel for us to use. Something like:

```java
public PineconeConnection(PineconeConfig config) {
  if (config.getManagedChannel() != null) {
    channel = config.getManagedChannel()
  } else {
    if (config.getHost() == null) {
       // throws exception
    }
    channel = buildChannel(config.getHost());
  }

   initialize();
}
```

Reply:
We already have the constructor of type `PineconeConnection(PineconeConfig config, String indexName)` so what you're suggesting is to delete the constructor `PineconeConfig(String apiKey, String indexName)`
The reason it was added in the first place is because I was trying to replicate the python and node sdk behavior of being able to make data plane calls with just apiKey and indexName. 

So as a next commit, I'll delete `PineconeConfig(String apiKey, String indexName)` constructor and add `PineconeConfig(PineconeConfig config)`.

Reply:
Yeah, I see now. Sorry I overlooked the `PineconeConnection(PineconeConfig config, String indexName)` that's already implemented. The way github renders the diff is a bit hard to follow.